macros/contexts/contextBoolean.pl

```diff
@@ -0,0 +1,160 @@
+## contextBoolean.pl
+
+sub _contextBoolean_init { context::Boolean::Init() }
+
+package context::Boolean;
+
+sub Init {
+	my $context = $main::context{Boolean} = Parser::Context->getCopy("Numeric");
+	$context->{name} = "Boolean";
+	$context->{value}{Formula} = "context::Boolean::Formula";
+
+	## Disable unwanted operations and functions
+	$context->functions->disable('All');
+	#    $context->parens->undefine('<');
+	$context->operators->undefine('-', '/', ' /', '/ ', '//', '^', '**', 'u+', '!');
+	$context->operators->redefine('><');
+
+	## Re-enable necessary operations
+	$context->operators->set(
+		'+'   => { class => "Parser::BOP::or",  string => ' or ',  TeX => '\vee ',   perl => '||' },
+		'or'  => { class => "Parser::BOP::or",  string => ' or ',  TeX => '\vee ',   perl => '||' },
+		'and' => { class => "Parser::BOP::and", string => ' and ', TeX => '\wedge ', perl => '&&' },
+		'*'   => { class => "Parser::BOP::and", string => ' and ', TeX => '\wedge ', perl => '&&' },
+		' *'  => { class => "Parser::BOP::and", string => ' and ', TeX => '\wedge ', perl => '&&' },
+		'* '  => { class => "Parser::BOP::and", string => ' and ', TeX => '\wedge ', perl => '&&' },
+		# implicit multiplication must have string '*' for Parser::Op (L#314)
+		' '   => { class => "Parser::BOP::and", string => ' *',    TeX => '\wedge ' },
+		'u-'  => { class => "Parser::UOP::not", string => '-',     TeX => '\mathord{\sim}' },
+		'><'  => { class => "Parser::BOP::xor", string => ' xor ', TeX => '\oplus ' },
+		'xor' => { class => "Parser::BOP::xor", string => ' xor ', TeX => '\oplus ' },
+	);
+
+	## Set default variables 'p' and 'q'
+	$context->variables->are(
+		p => 'Real',
+		q => 'Real',
+	);
+}
+
+## Subclass Value::Formula for boolean formulas
+package context::Boolean::Formula;
+our @ISA = ('Value::Formula');
+
+sub new {
+	my $class = shift;
+	my $self  = $class->SUPER::new(@_);
+
+	my @variables = $self->{context}->variables->names;
+	my @points;
+	my @values;
+
+	my $f = $self->{f} = $self->perlFunction(undef, \@variables);
+
+	foreach my $combination (0 .. 2**@variables - 1) {
+		my @point = map { $combination & 2**$_ ? 1 : 0 } (0 .. $#variables);
+		my $value = &$f(@point) ? 1 : 0;
+		push @points, \@point;
+		push @values, $value;
+	}
+
+	$self->{test_points} = \@points;
+	$self->{test_values} = \@values;
+
+	return $self;
+}
+
+package Parser::BOP::or;
+our @ISA = qw(Parser::BOP);
+
+sub _check {
+	my $self = shift;
+	# Add logic to check compatibility for logical 'or'
```

Comment:
I'm curious about this -- but didn't quite see how to implement such an idea. I'm thinking that I would want to correlate this with the variables added to this context as well... 

Something like:
```perl
$context->variables->are( 
  p => { isValue => 1, type => Value::Type('Boolean', 1) },
  q => { isValue => 1, type => Value::Type('Boolean', 1) },
);

# totally unsure about what follows... {value} or {parser}?
$context->{parser}{Boolean} = "context::Boolean::Variable";
$context->{parser}{Number} = "context::Boolean::Number";
```

On a related(?) note, I needed to add `context::Boolean::Formula::typeRef` (otherwise ->cmp died gruesomely), which I set to return `$Value::Type{number}` -- and this "worked", but working with this suggestion, I would also want to alter this as well...